### ISSUE TITLE
How to configure the amplify tool to use configured Service Role when performing CI/CD backend deploy

### ISSUE THREAD

**Issue by jleskovar-tyro** [aws-amplify/amplify-console#2396]
**Please describe which feature you have a question about?**
Backend deploy through CI/CD in Amplify Console

**Provide additional details**
Backend deploy hangs on prompt for auth method, using `amplifyPush --simple` and the latest amplify-cli. I've created a service role for the deploy, and have already configured this in app settings. However, it seems like the amplify cli tool does not use this service role when running. Instead, the backend deploy hangs with the following output:
```
                                 # Starting phase: build
2021-11-20T22:15:03.267Z [INFO]: ? Select the authentication method you want to use: (Use arrow keys)
                                 ❯ AWS profile
                                 AWS access keys
```

**What AWS Services are you utilizing?**
Amplify-CLI 7.4.4 (latest), amplifyPush

**Provide additional details e.g. code snippets**
amplify.yml:
```
version: 1
 backend:
   phases:
     build:
       commands:
         - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn generate
  artifacts:
    # IMPORTANT - Please verify your build output directory
    baseDirectory: dist
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

**Comment by jleskovar** [comment]
I think I fixed my issue - though not sure how. I ended up blowing away my backend and amplify directory, and basically re-initialising everything from scratch, using the CLI from my local dev machine. Now when I pin the amplify-cli package to latest in Build settings, I no longer have the issue with the Auth method prompt. I think maybe my repo got into a weird state since I was mixing CLI usage with the Admin UI?
